CMakeLists.txt

```diff
@@ -1,35 +1,62 @@
+cmake_minimum_required(VERSION 3.25)
 project(bncsutil)
-cmake_minimum_required(VERSION 3.2)
 
 set(CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR} "${PROJECT_SOURCE_DIR}/CMake/Modules")
 message(${CMAKE_BINARY_DIR})
 
 SET(VERSION_MAJOR "1")
 SET(VERSION_MINOR "4")
-SET(VERSION_PATCH "1")
-
-file(GLOB SOURCE
-    "src/bncsutil/*.cpp"
-    "src/bncsutil/*.c"
+SET(VERSION_PATCH "2")
+
+add_library(bncsutil SHARED)
+
+set(HEADERS
+    "src/bncsutil/bncsutil.h"
+    "src/bncsutil/bsha1.h"
+    "src/bncsutil/buffer.h"
+    "src/bncsutil/cdkeydecoder.h"
+    "src/bncsutil/checkrevision.h"
+    "src/bncsutil/decodekey.h"
+    "src/bncsutil/file.h"
+    "src/bncsutil/keytables.h"
+    "src/bncsutil/libinfo.h"
+    "src/bncsutil/ms_stdint.h"
+    "src/bncsutil/mutil.h"
+    "src/bncsutil/mutil_types.h"
+    "src/bncsutil/nls.h"
+    "src/bncsutil/oldauth.h"
+    "src/bncsutil/pe.h"
+    "src/bncsutil/sha1.c"
+    "src/bncsutil/sha1.h"
+    "src/bncsutil/stack.h"
 )
-file(GLOB HEADERS
-    "src/bncsutil/*.h"
+
+set(SOURCES
+    "src/bncsutil/bsha1.cpp"
+    "src/bncsutil/cdkeydecoder.cpp"
+    "src/bncsutil/checkrevision.cpp"
+    "src/bncsutil/decodekey.cpp"
+    "src/bncsutil/file.cpp"
+    "src/bncsutil/libinfo.cpp"
+    "src/bncsutil/oldauth.cpp"
 )
 
-add_library(${PROJECT_NAME} ${SOURCE} ${HEADERS})
+target_sources(bncsutil PRIVATE ${SOURCES} ${HEADERS})
```

Comment:
**Inconsistency:** Seemed to have removed `${PROJECT_NAME}` here but not in other places like line 82.

Should the lines [44, 53, 59, 70, 71, 75, 76, 79] replace `bncsutil` with `${PROJECT_NAME}`?

Reply:
`${PROJECT_NAME}` was a supposedly a "smart" idea of mine in the past but I've never seen this in modern cmake in other projects. Fixed in https://github.com/BNETDocs/bncsutil/pull/18/commits/9fd0649220e8d1fc55e356dab8af5c8deb89841c